### ISSUE TITLE
Delete BatchNormalization layer causes loss cannot reduce properly

### ISSUE THREAD

**Issue by ysfxgjy** [pk00095/transfer-learning-with-tensorflow#1]
Thank you for the code. The modified add_final_training_ops function is very helpful. I got 70% accuracy while training, however, when I use the retraied_graph.pb file for image prediction, the whole performance becomes very poor.

Another problem is when I deleted the BatchNormalization layer, the cross entropy will no longer reduce. 

I guess these two problems are related to the fact that the weight of the network is not properly saved or updated. Do you know where the problem is? Thank you.

**Train logs**
> INFO:tensorflow:2018-07-26 15:28:02.482981: Step 0: Train accuracy = 4.0%
INFO:tensorflow:2018-07-26 15:28:02.482981: Step 0: Cross entropy = 5.147309
INFO:tensorflow:2018-07-26 15:28:10.097577: Step 0: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:17.475636: Step 10: Train accuracy = 2.0%
INFO:tensorflow:2018-07-26 15:28:17.475636: Step 10: Cross entropy = 5.147165
INFO:tensorflow:2018-07-26 15:28:18.149421: Step 10: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:24.574369: Step 20: Train accuracy = 1.0%
INFO:tensorflow:2018-07-26 15:28:24.575368: Step 20: Cross entropy = 5.147106
INFO:tensorflow:2018-07-26 15:28:25.267146: Step 20: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:32.510834: Step 30: Train accuracy = 2.0%
INFO:tensorflow:2018-07-26 15:28:32.511836: Step 30: Cross entropy = 5.147902
INFO:tensorflow:2018-07-26 15:28:33.132634: Step 30: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:39.460614: Step 40: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:28:39.460614: Step 40: Cross entropy = 5.147810
INFO:tensorflow:2018-07-26 15:28:40.117404: Step 40: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:46.508364: Step 50: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:28:46.508364: Step 50: Cross entropy = 5.147882
INFO:tensorflow:2018-07-26 15:28:47.131161: Step 50: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:53.146240: Step 60: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:28:53.147239: Step 60: Cross entropy = 5.147972
INFO:tensorflow:2018-07-26 15:28:53.739051: Step 60: Validation accuracy = 2.0% (N=100)
INFO:tensorflow:2018-07-26 15:28:59.649185: Step 70: Train accuracy = 1.0%
INFO:tensorflow:2018-07-26 15:28:59.649185: Step 70: Cross entropy = 5.145600
INFO:tensorflow:2018-07-26 15:29:00.431913: Step 70: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:06.365018: Step 80: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:29:06.366020: Step 80: Cross entropy = 5.147554
INFO:tensorflow:2018-07-26 15:29:06.990790: Step 80: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:12.831961: Step 90: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:29:12.831961: Step 90: Cross entropy = 5.148786
INFO:tensorflow:2018-07-26 15:29:13.441760: Step 90: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:19.363865: Step 100: Train accuracy = 1.0%
INFO:tensorflow:2018-07-26 15:29:19.364873: Step 100: Cross entropy = 5.146307
INFO:tensorflow:2018-07-26 15:29:19.986665: Step 100: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:25.749834: Step 110: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:29:25.749834: Step 110: Cross entropy = 5.146955
INFO:tensorflow:2018-07-26 15:29:26.313644: Step 110: Validation accuracy = 2.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:31.962837: Step 120: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:29:31.962837: Step 120: Cross entropy = 5.147587
INFO:tensorflow:2018-07-26 15:29:32.528656: Step 120: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:38.202848: Step 130: Train accuracy = 1.0%
INFO:tensorflow:2018-07-26 15:29:38.203855: Step 130: Cross entropy = 5.146802
INFO:tensorflow:2018-07-26 15:29:38.768664: Step 130: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:44.807890: Step 140: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:29:44.807890: Step 140: Cross entropy = 5.148148
INFO:tensorflow:2018-07-26 15:29:45.344713: Step 140: Validation accuracy = 1.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:50.953925: Step 150: Train accuracy = 0.0%
INFO:tensorflow:2018-07-26 15:29:50.953925: Step 150: Cross entropy = 5.146530
INFO:tensorflow:2018-07-26 15:29:51.517739: Step 150: Validation accuracy = 0.0% (N=100)
INFO:tensorflow:2018-07-26 15:29:57.135949: Step 160: Train accuracy = 0.0%
>

**Comment by ysfxgjy** [comment]
After using retrain.py from the latest commit and setting is_training flag properly, I can get right result from retrained_graph.pb. However, after deleting the BatchNormalization layer, the cross entropy still can't reduce.

**Comment by pk00095** [comment]
@ysfxgjy 

```
    with tf.name_scope('hidden2_weights'):
      hidden2_weight = tf.truncated_normal([1024, 1024], stddev=0.001)
      layer2_weights = tf.Variable(hidden2_weight,name='hidden2_weights')
      variable_summaries(layer2_weights)
    with tf.name_scope('hidden2_biases'):
      layer2_biases = tf.Variable(tf.zeros([1024]),name='hidden2_biases')
      variable_summaries(layer2_biases)
    with tf.name_scope('hidden2_Wx_plus_b'):
      logits_hidden2 = tf.matmul(dropped_hidden1,layer2_weights)+layer2_biases
      tf.summary.histogram('hidden2_pre_activations',logits_hidden2)

    with tf.name_scope('hidden2_activation'):
      hidden2_activated=tf.nn.relu(logits_hidden2,name='hiddenLayer2_activation')
      tf.summary.histogram('hiddenLayer2_Activation',hidden2_activated)
    with tf.name_scope('hidden2_dropout'):
      dropped_hidden2=tf.layers.dropout(hidden2_activated,rate=0.5,training=True,name='hidden2_dropout_layer')
     variable_summaries(dropped_hidden2)
```

These are the changes you need to do if you want to remove batch normalization 

1. add the bias(_while using batchnorm we don't add biases explicitly because this is done internally_)
2. feed the logit to next activation instead of batch normalised logit.
